code/modules/luck/luck.dm

```diff
@@ -0,0 +1,149 @@
+//Luckiness and unluckiness
+
+/mob/proc/luck()
+	var/luck = 0
+	if(base_luck)
+		luck += base_luck.base_luck()
+	//Adjust based on borne items
+	luck += borne_item_luckiness()
+	return luck
+
+/mob
+	var/datum/luckiness/base_luck
+
+/mob/proc/luck_adjust(var/luckchange, var/temporary = FALSE)
+	if(!base_luck)
+		base_luck = new /datum/luckiness
+	if(temporary)
+		base_luck.temporary_luckiness += luckchange
+	else
+		base_luck.permanent_luckiness += luckchange
+
+/mob/proc/borne_item_luckiness() //Check all inventory items for (un)luckiness.
+	var/total_item_luckiness = 0
+	var/list/equipped_items = get_equipped_items()
+	while(null in equipped_items)
+		equipped_items -= null
+	var/list/held_or_implanted_items = contents - equipped_items
+	//check hand slot items and implanted items
+	if(held_or_implanted_items.len)
+		for(var/i in 1 to held_or_implanted_items.len)
```

Comment:
This and all other similar loops below should just be like `for(var/type/thing in list)`